src/engine/strat_engine/backstore/shared.rs

```diff
@@ -58,34 +58,21 @@ pub fn metadata_to_segment(
 // OR one should go away entirely in solution to:
 // https://github.com/stratis-storage/stratisd/issues/762.
 pub fn coalesce_blkdevsegs(left: &[BlkDevSegment], right: &[BlkDevSegment]) -> Vec<BlkDevSegment> {
-    if left.is_empty() {
-        return right.to_vec();
-    }
-    if right.is_empty() {
-        return left.to_vec();
-    }
-
-    let mut segments = Vec::with_capacity(left.len() + right.len());
-    segments.extend_from_slice(left);
-
-    // Last existing and first new may be contiguous.
-    let coalesced = {
-        let right_first = right.first().expect("!right.is_empty()");
-        let left_last = segments.last_mut().expect("!left.is_empty()");
-        if left_last.uuid == right_first.uuid
-            && (left_last.segment.start + left_last.segment.length == right_first.segment.start)
-        {
-            left_last.segment.length += right_first.segment.length;
-            true
-        } else {
-            false
-        }
-    };
-
-    if coalesced {
-        segments.extend_from_slice(&right[1..]);
-    } else {
-        segments.extend_from_slice(right);
-    }
-    segments
+    left.iter().chain(right.iter()).cloned().fold(
```

Comment:
I was looking at that and I wasn't entirely sure how to best approach it because these two methods operate on totally different data types. Are you hoping to abstract the logic across both data types?

Reply:
They do! If you want to take the shortest route on this, you can just remove the FIXME's from the header comments for both methods and let them go their separate ways.